applications/minotari_node/src/commands/command/test_peer_liveness.rs

```diff
@@ -0,0 +1,216 @@
+//  Copyright 2022, The Tari Project
+//
+//  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+//  following conditions are met:
+//
+//  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+//  disclaimer.
+//
+//  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+//  following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+//  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+//  products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::{fs, fs::OpenOptions, io::Write, path::PathBuf, process, time::Instant};
+
+use anyhow::Error;
+use async_trait::async_trait;
+use chrono::Local;
+use clap::Parser;
+use minotari_app_utilities::utilities::UniPublicKey;
+use tari_comms::{
+    multiaddr::Multiaddr,
+    net_address::{MultiaddressesWithStats, PeerAddressSource},
+    peer_manager::{NodeId, Peer, PeerFeatures, PeerFlags},
+};
+use tari_p2p::services::liveness::LivenessEvent;
+use tokio::{sync::watch, task};
+
+use super::{CommandContext, HandleCommand};
+
+/// Adds a peer
+#[derive(Debug, Parser)]
+pub struct ArgsTestPeerLiveness {
+    /// The public key of the peer to be tested
+    public_key: UniPublicKey,
+    /// The address of the peer to be tested
+    address: Multiaddr,
+    /// Auto exit the base node after test
+    exit: Option<bool>,
+    /// Write the responsiveness result to file - results will be written to
+    /// 'peer_liveness_test.log'
+    output_to_file: Option<bool>,
+    /// Start with a new log file
+    refresh_file: Option<bool>,
+    /// Optional output directory (otherwise current directory will be used)
+    output_directory: Option<PathBuf>,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+enum PingResult {
+    Initial,
+    Success,
+    Fail,
+}
+
+#[allow(clippy::too_many_lines)]
```

Comment:
Can you try and split this into multiple functions?
then we don't need the clippy bypass here and the function is smaller and easier to read.